cirq-core/cirq/devices/fidelity_test.py

```diff
@@ -0,0 +1,210 @@
+        cirq.Moment([cirq.measure(qubits[0], key='q0'), cirq.measure(qubits[1], key='q1')]),
+    )
+
+    assert_equivalent_op_tree(expected_circuit, noisy_circuit)
+
+
+def test_depolarization_error():
+    pauli_error = 0.1
+
+    # Create qubits and circuit
+    qubits = [cirq.LineQubit(0), cirq.LineQubit(1)]
+    circuit = cirq.Circuit(
+        cirq.Moment([cirq.X(qubits[0])]),
+        cirq.Moment([cirq.CNOT(qubits[0], qubits[1])]),
+        cirq.Moment([cirq.H(qubits[1])]),
+        cirq.Moment([cirq.measure(qubits[0], key='q0'), cirq.measure(qubits[1], key='q1')]),
+    )
+
+    # Create noise model from Fidelity object with specified noise
+    f = Fidelity(pauli_error=pauli_error)
+    noise_model = NoiseModelFromFidelity(f)
+
+    noisy_circuit = cirq.Circuit(noise_model.noisy_moments(circuit, qubits))
+
+    # Insert expected channels to circuit
+    expected_circuit = cirq.Circuit(
+        cirq.Moment([cirq.X(qubits[0])]),
+        cirq.Moment([cirq.depolarize(pauli_error / 3).on_each(qubits)]),
+        cirq.Moment([cirq.CNOT(qubits[0], qubits[1])]),
+        cirq.Moment([cirq.depolarize(pauli_error / 3).on_each(qubits)]),
+        cirq.Moment([cirq.H(qubits[1])]),
+        cirq.Moment([cirq.depolarize(pauli_error / 3).on_each(qubits)]),
+        cirq.Moment([cirq.measure(qubits[0], key='q0'), cirq.measure(qubits[1], key='q1')]),
+    )
+
+    assert_equivalent_op_tree(expected_circuit, noisy_circuit)
+
+
+def test_ampl_damping_error():
+    t1 = 200.0
+
+    # Create qubits and circuit
+    qubits = [cirq.LineQubit(0), cirq.LineQubit(1)]
+    circuit = cirq.Circuit(
+        cirq.Moment([cirq.X(qubits[0])]),
+        cirq.Moment([cirq.CNOT(qubits[0], qubits[1])]),
+        cirq.Moment([cirq.FSimGate(5 * np.pi / 2, np.pi).on_each(qubits)]),
+        cirq.Moment([cirq.measure(qubits[0], key='q0'), cirq.measure(qubits[1], key='q1')]),
+    )
+
+    # Create noise model from Fidelity object with specified noise
+    f = Fidelity(t1=t1)
+    noise_model = NoiseModelFromFidelity(f)
+
+    noisy_circuit = cirq.Circuit(noise_model.noisy_moments(circuit, qubits))
+
+    # Insert expected channels to circuit
+    expected_circuit = cirq.Circuit(
+        cirq.Moment([cirq.X(qubits[0])]),
+        cirq.Moment([cirq.amplitude_damp(1 - np.exp(-25.0 / t1)).on_each(qubits)]),
+        cirq.Moment([cirq.CNOT(qubits[0], qubits[1])]),
+        cirq.Moment([cirq.amplitude_damp(1 - np.exp(-25.0 / t1)).on_each(qubits)]),
+        cirq.Moment([cirq.FSimGate(np.pi / 2, np.pi).on_each(qubits)]),
+        cirq.Moment([cirq.amplitude_damp(1 - np.exp(-12.0 / t1)).on_each(qubits)]),
+        cirq.Moment([cirq.measure(qubits[0], key='q0'), cirq.measure(qubits[1], key='q1')]),
+    )
+
+    assert_equivalent_op_tree(expected_circuit, noisy_circuit)
+
+
+def test_combined_error():
+    # Helper function to calculate pauli error from depolarization
+    def pauli_error_from_depolarization(pauli_error, t1, duration):
```

Comment:
This conversion seems like it would be useful outside of tests - can it be moved to `fidelity.py`?